src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -0,0 +1,269 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static class LambdaUtilities
+    {
+        /// <summary>
+        /// Returns true if the specified node represents a lambda.
+        /// </summary>
+        public static bool IsLambda(SyntaxNode node)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.LetClause:
+                case SyntaxKind.WhereClause:
+                case SyntaxKind.AscendingOrdering:
+                case SyntaxKind.DescendingOrdering:
+                case SyntaxKind.SelectClause:
```

Comment:
I love subtle details! /s

In any case, it's ok to claim an expression belongs to a lambda even if it doesn't. It's not ideal, more rude edits might be reported, but it's not critical. I'll file a bug to handle these subtle cases.